src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -845,18 +847,260 @@ public Document GetRevisionBefore(DocumentsOperationContext context, string id,
+                    }
+
+                    tx.Commit();
+                }
+
+                list.Clear();
+            }
+        }
+
+        private bool PrepareRevertedRevisions(DocumentsOperationContext writeCtx, DateTime before, DateTime dateToBreak, RevertResult result, List<Document> list, HashSet<string> scannedIds)
+        {
+            using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext readCtx))
+            using (readCtx.OpenReadTransaction())
+            {
+                var revisions = new Table(RevisionsSchema, readCtx.Transaction.InnerTransaction);
+                foreach (var tvr in revisions.SeekBackwardFromLast(RevisionsSchema.FixedSizeIndexes[AllRevisionsEtagsSlice],
+                    result.Progress.ScannedRevisions))
+                {
+                    result.Progress.ScannedRevisions++;
+                    var id = TableValueToString(readCtx, (int)RevisionsTable.LowerId, ref tvr.Reader);
+
+                    if (scannedIds.Add(id) == false)
+                        continue;
+
+                    result.Progress.ScannedDocuments++;
+
+                    if (_documentsStorage.ConflictsStorage.HasConflictsFor(readCtx, id))
+                    {
+                        result.Progress.Warn(id, "The document is conflicted and wouldn't be reverted.");
+                        continue;
+                    } 
+
+                    var date = TableValueToDateTime((int)RevisionsTable.LastModified, ref tvr.Reader);
+                    if (date < dateToBreak)
+                        break;
+
+                    RestoreRevision(readCtx, writeCtx, id, before, result ,list);
+                    if (readCtx.AllocatedMemory + writeCtx.AllocatedMemory > SizeLimit)
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        }
+
+        private void RestoreRevision(DocumentsOperationContext readCtx, DocumentsOperationContext writeCtx, LazyStringValue id, DateTime before, RevertResult result, List<Document> list)
+        {
+            var revision = GetRevisionBefore(readCtx, id, before, out var foundAfter, out var wasConflicted);
+
+            if (foundAfter == false)
+                return; // no changes after the requested time or no revision was found
+
+            if (revision == null)
+            {
+                result.Progress.Warn(id, $"The oldest revision is after the '{before}'.");
+                return;
+            }
+
+            if (wasConflicted)
+            {
+                result.Progress.Warn(id, $"Skip revert, since the document was conflicted during '{before}'.");
+                return;
+            }
+
+            var collection = CollectionName.GetCollectionName(revision.Data);
+            var configuration = GetRevisionsConfiguration(collection);
+            if (configuration == _emptyConfiguration)
```

Comment:
I don't understand the logic here? 
Why do you care about the revisions configuration here?